tests/testutil/service.go

```diff
@@ -202,6 +203,76 @@ func (ts *TestService) GetRunningPods() string {
 	return out
 }
 
+func (ts *TestService) GetEndpointUrl() string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	url, err := ts.waitForEndpointDNS()
+	if err != nil {
+		ts.T.Fatalf("Failed to get the endpoint url:  %v", err.Error())
+		return ""
+	}
+	return url
+}
+
+func (ts *TestService) GetKubeEndpointUrlByRevision(revision string) string {
+	err := ts.reloadApp()
+	if err != nil {
+		ts.T.Fatalf("Failed to reload app:  %v", err.Error())
+		return ""
+	}
+	args := append([]string{"--format", "json",
+		fmt.Sprintf("%s:%s", strings.Split(ts.Name, ":")[0], revision)})
+	out, err := RioCmd("inspect", args)
+	ts.Service = riov1.Service{}
+	if err := json.Unmarshal([]byte(out), &ts.Service); err != nil {
+		return ""
+	}
+	args = append([]string{"get", "service.rio.cattle.io",
+		"-n", testingNamespace,
+		ts.Service.Name,
+		"-o", `jsonpath="{.status.endpoints[0]}"`})
+	url, err := KubectlCmd(args)
+	if err != nil {
+		ts.T.Fatalf("Failed to endpoint url:  %v", err.Error())
+		return ""
+	}
+	url = url[1 : len(url)-1] // remove double quotes from output
```

Comment:
Can we use `strings.Replace(url, "\"", "", -1)` instead or will that mess with other quotes ? We should at least ensure that `url != ""` before doing this or it will panic.